### ISSUE TITLE
Make wdio cli tool more powerful

### ISSUE THREAD

**Issue by christian-bromann** [webdriverio/webdriverio#3915]
**Is your feature request related to a problem? Please describe.**
For newcomers it is not straight forward to handle plugins in WebdriverIO. We should make the CLI tool more powerful to make the setup easier.

**Describe the solution you'd like**
Add an `install` command that allows things like:
```sh
$ wdio install spec reporter
$ wdio install sauce service
```
it should automatically add the service to the config. If no config in the execution directory is found it should ask for the file or allow to skip and exit with an information what needs to be added to the config. Maybe as a first step we can do latter and add the automatic code change later on.

**Describe alternatives you've considered**
None

**Additional context**
None

**Comment by baruchvlz** [comment]
It would make more sense to have the type of thing installed first.

```shell
$ wdio install reporter <name>
$ wdio install service <name>
```

Could also be short handed:

```shell
$ wdio i r <name>
$ wdio i s <name>
```

As for handling the missing config, maybe give the user 2 options, 1) use a default configuration (which would mean a local runner and the minimum configuration needed for wdio to run), or 2) go through the config init process. In any case, the reporter or service should be installed after.

I'm guessing that with this tool we also want to install the necessary package from NPM, so we should handle errors. For example if trying to install a service that's for v4 and you're using v5, it shouldn't install and say that there's a version mismatch.

**Comment by CrispusDH** [comment]
I don't see profit from adding complexity by handling the missing basic config. As for me it would be enough to send message in this case (e.g. "Basic `wdio.config.js` is missing. Create it firstly.").

**Comment by baruchvlz** [comment]
@CrispusDH Yea, that's a fair point, creating a basic configuration automatically could lead to people being a bit confused.

We could handle it more gracefully:

```
$ wdio install service sauce
Error: Could not install "sauce" service due to missing configuration.
Would you like to create one?(Y/n)
```

If `Y` then we just start the `wdio config` command. And afterwards we can ask if they want their service installed.

**Comment by duderonomy** [comment]
Does `wdio config` perform a similar to job as this proposed install command? 
What does one method accomplish that the other does not?

**Comment by baruchvlz** [comment]
@duderonomy That's a fair point, the `wdio config` command does have the ability to install reporters, services, and frameworks but that will only work on the initial setup of the WebdriverIO project. If you wanted to install a new reporter, for example, you will have to do this manually (verify if it's a supported reporter, install it, update the config, etc.). Granted you could run the config again, but this will overwrite your configuration, with the proposed `install` we can delegate these checks, installations, and updates to WDIO.

In the future there will be [more features](https://github.com/webdriverio/webdriverio/pull/4010#issuecomment-495597240) added to the command that allow for easier installation and configuration.

**Comment by christian-bromann** [comment]
implemented in #4010
